workers/datasets_based/src/datasets_based/workers/split_names.py

```diff
@@ -0,0 +1,132 @@
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
(just curious) why did you choose to add a new /split-names while you could have just extended /splits to support a config= query parameter ?

Reply:
I find it a lot easier to create a new endpoint, populate its cache, then switch ([blue/ green method](https://www.redhat.com/en/topics/devops/what-is-blue-green-deployment)) than having to manage two different logics at the same time.